mmpose/datasets/datasets/top_down/topdown_coco_dataset.py

```diff
@@ -359,14 +371,15 @@ def evaluate(self, outputs, res_folder, metric='mAP', **kwargs):
                 # rescoring
                 n_p['score'] = kpt_score * box_score
 
-            nms = soft_oks_nms if self.soft_nms else oks_nms
-            keep = nms(list(img_kpts), oks_thr, sigmas=self.sigmas)
-
-            if len(keep) == 0:
-                oks_nmsed_kpts.append(img_kpts)
+            if self.use_nms:
+                nms = soft_oks_nms if self.soft_nms else oks_nms
+                keep = nms(list(img_kpts), oks_thr, sigmas=self.sigmas)
+                if len(keep) == 0:
+                    oks_nmsed_kpts.append(img_kpts)
```

Comment:
If img_kpts is non empty, then the result of nms could still be empty, right or wrong?

Reply:
The result of nms won't be empty as long as img_kpts has at least one element in it.  But this operation has been there in previous versions of mmpose, so I just kept it. 
May I remove these lines?